it_hardware/data/ir_model_fields.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="new_date_lot_serial_x_warranty_date" model="ir.model.fields">
+        <field name="name">x_warranty_date</field>
+        <field name="model_id" ref="stock.model_stock_lot"/>
+        <field name="ttype">date</field>
+        <field name="field_description">Warranty Date</field>
+    </record>
```

Comment:
Couldn't it be a computed field, based on the information from the product?
As soon as the lot is delivered, the warranty date is set